lib/tbot/config/config.go

```diff
@@ -207,6 +207,34 @@ func (conf *BotConfig) CipherSuites() []uint16 {
 	return utils.DefaultCipherSuites()
 }
 
+func (conf *BotConfig) UnmarshalYAML(node *yaml.Node) error {
+	// Wrap conf in an anonymous struct to avoid having the deprecated field on
+	// the BotConfig or CredentialLifetime structs, and keep it purely a config
+	// file parsing concern.
+	//
+	// The type alias prevents infinite recursion by obscuring UnmarshalYAML.
+	type alias BotConfig
+	output := struct {
+		*alias                   `yaml:",inline"`
```

Comment:
Do we actually need to embed the BotConfig itself? Would an anonymous struct with just the DeprecatedCertificateTTL field work here instead?

Reply:
Ah right - that makes sense. Fine by me as it is.